File: python/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    # O(n) time | O(1) space
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 0
        val, count = nums[0], 0
        for num in nums:
            if num == val:
                count += 1
            else:
                val = num
                count = 1
            if count <= 2:
                nums[k] = num
                k += 1
        return k

File: c++/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    // O(n) time, O(1) space - where n is the size of the array
    int removeDuplicates(vector<int>& nums) {
        int k = 0;
        int val = nums[0], count = 0;
        for (int num : nums) {
            if (num == val) {
                ++count;
            } else {
                val = num;
                count = 1;
            }
            if (count <= 2) {
                nums[k++] = num;
            } 
        }
        return k;
    }
};